SeaPublicWebsite/Models/EnergyEfficiency/EmailAddressViewModel.cs

```diff
@@ -15,8 +15,10 @@ public class EmailAddressViewModel
         public string EmailAddress { get; set; }
         
         public string Reference { get; set; }
-        public bool Change { get; set; }
+        public PageName? EntryPoint { get; set; }
 
         public bool IsRequiredEmailAddress => HasEmailAddress == QuestionOptions.HasEmailAddress.Yes;
+        
+        public string BackLink { get; set; }
```

Comment:
Sounds good. It may get a bit tricky right at the end of the journey (or when presenting the results), but for now the changes don't affect that side.